src/wkwebview/class/url_scheme_handler.rs

```diff
@@ -0,0 +1,278 @@
+// Copyright 2020-2024 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+use std::borrow::Cow;
+use std::ffi::c_void;
+use std::ptr::NonNull;
+use std::slice;
+
+use http::header::{CONTENT_LENGTH, CONTENT_TYPE};
+use http::{Request, Response as HttpResponse, StatusCode, Version};
+use objc2::rc::Retained;
+use objc2::runtime::{AnyClass, AnyObject, ClassBuilder, ProtocolObject};
+use objc2::ClassType;
+use objc2_foundation::NSObjectProtocol;
+use objc2_foundation::{
+  NSData, NSHTTPURLResponse, NSMutableDictionary, NSObject, NSString, NSURL, NSUUID,
+};
+use objc2_web_kit::{WKURLSchemeHandler, WKURLSchemeTask};
+
+use crate::wkwebview::WEBVIEW_IDS;
+use crate::{RequestAsyncResponder, WryWebView};
+
+pub fn create(name: &str) -> &AnyClass {
+  unsafe {
+    let scheme_name = format!("{}URLSchemeHandler", name);
+    let cls = ClassBuilder::new(&scheme_name, NSObject::class());
+    match cls {
+      Some(mut cls) => {
+        cls.add_ivar::<*mut c_void>("function");
+        cls.add_ivar::<u32>("webview_id");
+        cls.add_method(
+          objc2::sel!(webView:startURLSchemeTask:),
+          start_task as extern "C" fn(_, _, _, _),
+        );
+        cls.add_method(
+          objc2::sel!(webView:stopURLSchemeTask:),
+          stop_task as extern "C" fn(_, _, _, _),
+        );
+        cls.register()
+      }
+      None => AnyClass::get(&scheme_name).expect("Failed to get the class definition"),
+    }
+  }
+}
+
+// Task handler for custom protocol
+extern "C" fn start_task(
+  this: &AnyObject,
+  _sel: objc2::runtime::Sel,
+  webview: *mut WryWebView,
+  task: *mut ProtocolObject<dyn WKURLSchemeTask>,
```

Comment:
Nit: I _think_ you should be able to take references in these functions?

```suggestion
  webview: &WryWebView,
  task: &ProtocolObject<dyn WKURLSchemeTask>,
```